lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -616,3 +616,37 @@ OpFoldResult AsNonPassivePrimOp::fold(ArrayRef<Attribute> operands) {
 
   return {};
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// Connection like things
+///////////////////////////////////////////////////////////////////////////////
+void PartialConnectOp::getCanonicalizationPatterns(
+    OwningRewritePatternList &results, MLIRContext *context) {
+  struct Truncater final : public OpRewritePattern<PartialConnectOp> {
+    using OpRewritePattern::OpRewritePattern;
+
+    LogicalResult matchAndRewrite(PartialConnectOp op,
+                                  PatternRewriter &rewriter) const override {
+      auto destType =
+          op.getOperand(0).getType().cast<FIRRTLType>().getPassiveType();
+      auto srcType = op.getOperand(1).getType().cast<FIRRTLType>();
+      if (destType == srcType)
+        return failure();
+
+      auto srcWidth = srcType.getBitWidthOrSentinel();
```

Comment:
Plz add a comment about what this is doing so we can remember at a glance without studying the algorithm :-)